LEAF_Request_Portal/templates/view_reports.tpl

```diff
@@ -797,15 +797,58 @@ function openShareDialog() {
     });
 }
 
+function setExportFormat() {
+        if($('#shortenLink').css('display') === 'none') {
+            $('#exportFormat').html('?');
+        }
+        else {
+            $('#exportFormat').html('&');
+        }
+
+        switch($('#format').val().toLowerCase()) {
+            case 'json':
+            case 'bulkdatadownload':
+                $('#exportFormat').html('');
+                break;
+            default:
+                $('#exportFormat').append('format=' + $('#format').val());
+                $("#formatStatus").show().text("Format changed to " + $('#format').val());
+                $("#formatStatus").fadeOut(3000);
+                break;
+        }
+}
+
+let maxLimit = 10000;
+
+/**
+ * @param currentQuery
+ * @return currentQuery
+ */
+function setQueryLimit(currentQuery){
+    
+    switch($('#format').val().toLowerCase()) {
+        case 'bulkdatadownload':
+            delete currentQuery.limit;
+            delete currentQuery.limitOffset;
+            break;
+        default:
+            currentQuery.limit = maxLimit;
+            currentQuery.limitOffset = 0;
+            break;
+    }
+    return currentQuery;
+}
+
 function showJSONendpoint() {
-    var pwd = document.URL.substr(0,document.URL.lastIndexOf('?'));
-    leafSearch.getLeafFormQuery().setLimit(0, 10000);
-    var queryString = JSON.stringify(leafSearch.getLeafFormQuery().getQuery());
-    var jsonPath = pwd + leafSearch.getLeafFormQuery().getRootURL() + 'api/form/query/?q=' + queryString;
-    var powerQueryURL = '<!--{$powerQueryURL}-->' + window.location.pathname;
+    let pwd = document.URL.substr(0,document.URL.lastIndexOf('?'));
+    leafSearch.getLeafFormQuery().setLimit(0, maxLimit);
+    let currentQuery = leafSearch.getLeafFormQuery().getQuery();
+    let jsonPath = pwd + leafSearch.getLeafFormQuery().getRootURL() + 'api/form/query/?q=' + JSON.stringify(currentQuery);
+    let powerQueryURL = '<!--{$powerQueryURL}-->' + window.location.pathname;
+
 
     dialog_message.setTitle('Data Endpoints');
-    dialog_message.setContent('<p>This provides a live data source for custom dashboards or automated programs.</p><p><b>A configurable limit of 10,000 records has been preset</b>.</p><br />'
+    dialog_message.setContent('<p>This provides a live data source for custom dashboards or automated programs.</p><p><b>A configurable limit of '+maxLimit.toLocaleString('en-US')+' records has been preset</b>.</p><br />'
```

Comment:
Wouldn't we want to keep some limits in place? The bulk download made sense if some limits were in place, if we are removing that then we would want to not have that option either.

Reply:
I meant we don't need the preset 10,000 limit. Internally we'll still support the limit parameter since it's used as part of a "lazy loading" feature.

Reply:
Okay removing.